src/ui/Table/Paged.svelte

```diff
@@ -44,83 +76,85 @@
   function onNextPage() {
     if (page >= pagesAmount) page = maxPage
     else page++
-    onPageChange(page)
+    onPageChange(page, pageSize)
   }
 
   function onPrevPage() {
     if (page <= 1) page = 0
     else page--
-    onPageChange(page)
+    onPageChange(page, pageSize)
   }
+
+  onMount(() => {
+    if (hasMoreItems) {
+      onPageChange(0, pageSize)
+    }
+  })
 </script>
 
-<div class={className}>
-  <Table {...$$restProps} items={pageItems} offset={pageOffset} {applySort}>
-    <slot />
-  </Table>
-</div>
+<Table {...restProps} class={className} items={pageItems} offset={pageOffset} {applySort}>
+  <slot />
+</Table>
 
-<div class="paged row v-center mrg-l mrg--t">
+<section class="paged row v-center gap-l mrg-l mrg--t single-line {pagedClassName}">
   {#if rows.length > 1}
-    <Tooltip on="click" position="top" bind:isOpened={isPageSizeOpened}>
-      <div slot="trigger" class="rows btn-2 btn--s mrg-s mrg--r row v-center justify">
-        {pageSize} rows <Svg id="arrow-down" w="8" h="5" />
-      </div>
+    <Tooltip on="click" position="top" bind:isOpened={isPageSizeOpened} let:trigger>
+      <button use:trigger class="rows-trigger btn-2 btn--s row v-center gap-m">
+        {pageSize} rows
+        <Svg id="arrow-down" w="8" h="5" />
+      </button>
 
-      <div slot="tooltip" class="pages column">
+      <tooltip slot="tooltip" class="pages column">
         {#each rows as row}
           <button
             class="btn-ghost"
             class:active={row === pageSize}
             on:click={() => onPageSizeChange(row)}>{row} rows</button
           >
         {/each}
-      </div>
+      </tooltip>
     </Tooltip>
   {/if}
 
-  <span class="c-waterloo">Page</span>
-
-  <input
-    type="number"
-    class="input txt-center mrg-l mrg--l mrg--r"
-    min="1"
-    value={page + 1}
-    on:change={onPageInput}
-    max={pagesAmount}
-  />
-
-  <span class="c-waterloo"
-    >of {pagesAmount}
-    <span class="mrg-s mrg--l" />
-    ({items.length} rows total)</span
-  >
-
-  <button
-    class="btn-2 btn--s row hv-center mrg-a mrg--l"
-    class:disabled={page <= 0}
-    on:click={onPrevPage}
-  >
-    Prev
-    <Svg id="arrow-right" w="5" h="8" class="$style.left mrg-m mrg--l" />
-  </button>
-  <button
-    class="btn-2 btn--s row hv-center mrg-s mrg--l"
-    class:disabled={page >= maxPage}
-    on:click={onNextPage}
-  >
-    <Svg id="arrow-right" w="5" h="8" class="mrg-m mrg--r" />
-    Next
-  </button>
-</div>
+  <page-indicator class="row v-center gap-s c-waterloo">
+    <span>Page</span>
+
+    <input
+      type="number"
+      class="input txt-center"
+      min="1"
+      value={page + 1}
+      on:change={onPageInput}
+      max={pagesAmount}
+    />
+
+    <span>of {pagesAmount}</span>
```

Comment:
Total rows information should be displayed on desktop/tablets to give user clear information on the table's contents